### ISSUE TITLE
Assistance for binary for amd machine with ubuntu 23.04

### ISSUE THREAD

**Issue by pdgonzalez872** [rustprooflabs/pgdd#43]
Hi! I have an amd machine with 23.04 on it. I saw you released a binary for this for 22 a couple of weeks ago, not sure if I can do anything to help/vet a release for 23.04.

Thanks!

**Comment by rustprooflabs** [comment]
@pdgonzalez872 Thank you for the offer!  Would you be able to test using the Docker build process to build the installer for your system?  Knowing I can't possibly support every OS/Architecture, my hope is to improve the documentation so others can easily create their own installer when required.

I just merged in #44 that does two main things.  It adds a 23.04 `Dockerfile`  that should be building usable installers for your system.  There was only a tiny adjustment from the 22.04 version.  I also updated the docs, specifically the [Advanced Install](https://github.com/rustprooflabs/pgdd/blob/main/ADVANCED-INSTALL.md) section. The section on [customizing the Docker build system](https://github.com/rustprooflabs/pgdd/blob/main/ADVANCED-INSTALL.md#customize-docker-build-system) now shows how to hack that script to only run a single OS and a single Postgres version.  That should set you up with an installer like `pgdd_0.5.1_lunar_pg16_amd64.deb`, then the normal [install instructions](https://github.com/rustprooflabs/pgdd/blob/main/README.md#install-from-binary) should take care of the rest.

I **think** things should *just work* but we are dealing with computers here. :smile:   If the instructions don't make sense or you run into errors let me know!

**Comment by rustprooflabs** [comment]
@pdgonzalez872 I added Lunar (23.04) binaries to the [0.5.1](https://github.com/rustprooflabs/pgdd/releases/tag/0.5.1) release too.

**Comment by pdgonzalez872** [comment]
that's awesome! I'll give it a spin this weekend and report back!

**Comment by rustprooflabs** [comment]
I'm going to close this issue. Please report back here if you have further questions or run into any problems!

**Comment by pdgonzalez872** [comment]
Ryan, I have the original email as unread, I promise. I guess that’s an insight on how things have been in the last few weeks :)Thanks for looking into it, I’m sure it works as expected.Will let you know otherwise, in 2055 when I get to it :)On Nov 4, 2023, at 6:02 PM, Ryan Lambert ***@***.***> wrote:﻿
Closed #43 as completed.

**Comment by pdgonzalez872** [comment]
hi @rustprooflabs! Reporting back, things work!

I followed this (from the book :slightly_smiling_face:) :
![image](https://github.com/rustprooflabs/pgdd/assets/8906358/593894a5-5c81-4eb8-83e8-6be1a1c4c2d6)

I did the `wget` (adjusted for my version), then `dpkg`, worked well. I tried the query and things cried. I looked around and realized we need to create the extension (missing from the book):

```
(localhost 🐘) postgres@postgres=# SELECT s_name, v_name, description
    FROM dd.views
    WHERE s_name = 'dd'
    ORDER BY v_name
;
ERROR:  relation "dd.views" does not exist
LINE 2:     FROM dd.views
                 ^
Time: 0.460 ms
(localhost 🐘) postgres@postgres=# CREATE EXTENSION pgdd;
CREATE EXTENSION
Time: 35.435 ms
(localhost 🐘) postgres@postgres=# SELECT * FROM dd.database;
┌─────┬──────────┬─────────┬──────────────┬─────────────┬────────────────┬────────────┬───────────────┬─────────────────┐
│ oid │ db_name  │ db_size │ schema_count │ table_count │ size_in_tables │ view_count │ size_in_views │ extension_count │
╞═════╪══════════╪═════════╪══════════════╪═════════════╪════════════════╪════════════╪═══════════════╪═════════════════╡
│   5 │ postgres │ 16 MB   │            2 │           4 │ 7288 kB        │         11 │ 0 bytes       │               3 │
└─────┴──────────┴─────────┴──────────────┴─────────────┴────────────────┴────────────┴───────────────┴─────────────────┘
(1 row)

Time: 96.031 ms
(localhost 🐘) postgres@postgres=# SELECT s_name, v_name, description
    FROM dd.views
    WHERE s_name = 'dd'
;
┌────────┬────────────────────┬────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ s_name │       v_name       │                                            description                                             │
╞════════╪════════════════════╪════════════════════════════════════════════════════════════════════════════════════════════════════╡
│ dd     │ tables             │ Data dictionary view: Lists tables, excluding system tables.                                       │
│ dd     │ views              │ Data dictionary view: Lists views, excluding system views.                                         │
│ dd     │ schemas            │ Data dictionary view: Lists schemas, excluding system schemas.                                     │
│ dd     │ columns            │ Data dictionary view: Lists columns, excluding system columns.                                     │
│ dd     │ functions          │ Data dictionary view: Lists functions, excluding system functions.                                 │
│ dd     │ partition_parents  │ Data dictionary view: Lists parent partition tables with aggregate details about child partitions. │
│ dd     │ partition_children │ Data dictionary view: Lists individual partitions (children) of partitioned tables.                │
│ dd     │ database           │ Data dictionary view: Provides basic statistics for the current database.                          │
│ dd     │ index              │ Data dictionary view: Details about indexes in current database.                                   │
└────────┴────────────────────┴────────────────────────────────────────────────────────────────────────────────────────────────────┘
(9 rows)

Time: 5.081 ms
(localhost 🐘) postgres@postgres=# 

```

Thank you!
